src/main/java/com/example/TicketBookingApp/service/ReservationService.java

```diff
@@ -0,0 +1,139 @@
+package com.example.TicketBookingApp.service;
+
+import com.example.TicketBookingApp.configuration.TicketTypeConfiguration;
+import com.example.TicketBookingApp.domain.Reservation;
+import com.example.TicketBookingApp.domain.Screening;
+import com.example.TicketBookingApp.domain.Seat;
+import com.example.TicketBookingApp.domain.Ticket;
+import com.example.TicketBookingApp.domain.dto.ReservationRequest;
+import com.example.TicketBookingApp.domain.dto.ReservationResponse;
+import com.example.TicketBookingApp.domain.dto.TicketRequest;
+import com.example.TicketBookingApp.exception.*;
+import com.example.TicketBookingApp.repository.ReservationRepository;
+import com.example.TicketBookingApp.repository.ScreeningRepository;
+import com.example.TicketBookingApp.repository.SeatRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.util.ObjectUtils;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ReservationService {
+
+    private final ReservationRepository reservationRepository;
+    private final ScreeningRepository screeningRepository;
+    private final SeatRepository seatRepository;
+    private final TicketTypeConfiguration ticketTypeConfiguration;
+
+    private final int MAX_TIME_BEFORE_START_SCREENING = 15;
+    private final int DISTANCE_OF_ONE_PLACE = 1;
+    private final int DISTANCE_OF_TWO_PLACES = 2;
+
+    public ReservationResponse createReservation(ReservationRequest reservationRequest) {
+        Screening screening = screeningRepository
+                .findById(reservationRequest.getScreeningId())
+                .orElseThrow(ScreeningNotFoundException::new);
+
+        Reservation reservation = Reservation
+                .builder()
+                .date(screening.getDate())
+                .time(screening.getStartTime())
+                .screening(screening)
+                .personName(reservationRequest.getPersonName())
+                .personSurname(reservationRequest.getPersonSurname())
+                .expirationTime(screening.getStartTime().plus(Duration.ofMinutes(30)).atDate(screening.getDate()))
+                .build();
+
+        reservation.setTickets(reservationRequest
+                .getTicketRequests()
+                .stream()
+                .map(ticketRequest -> createTicketForReservation(ticketRequest, reservation))
+                .collect(Collectors.toList()));
+
+
+        reservation.setTotalAmountToPay((float) reservation
+                .getTickets()
+                .stream()
+                .mapToDouble(Ticket::getTicketPrice)
```

Comment:
to double czy float? czy może coś lepszego by się przydało?

Reply:
Jak było wspomniane wyżej, lepszym wyborem byłby _BigDecimal_